### PR TITLE
feat(audit-service): support for multiple actedon and actiongroup

### LINKED ISSUE #2115: Add the option to archive multiple acted on values at once
**Is your feature request related to a problem? Please describe.**

Right now the custom filter for archival accepts only string type value in actedOn field and there is no field for actionGroup here.
This restricts the user to archive only one entity at a time, increasing the number of '/archive' calls and hence increasing the files => which inturn affects the time taken by '/export' job since it has to get data one by one from each file and thus increasing latency.

**Describe the solution you'd like**
Here we can use a string array for actedOn and groupBy fields allowing user to archive all at once.

### PR DESCRIPTION
**Pull request by yeshamavani**
gh-2115

## Description

now the archive API accepts a list of actedOn and actionGroup

Fixes #2115 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?
Created a sample application to test this particular feature.

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/audit-service/src/services/job-processing.service.ts [hunk 1]
```diff
@@ -53,7 +58,10 @@ export class JobProcessingService {
             };
           }
           if (condition.actedOn) {
-            customFilter.actedOn = condition.actedOn;
+            customFilter.actedOn = this.getFilter(condition.actedOn);
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply services/audit-service/src/services/job-processing.service.ts]: created another function to improve

File: services/audit-service/src/utils/file-check.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  filterUsed: CustomFilter,
+  customFilter: CustomFilter,
+): boolean {
+  if (customFilter.entityId == null || filterUsed.entityId == null) {
+    return true;
+  }
+  if (filterUsed.entityId === customFilter.entityId) {
+    return true;
+  }
+  return false;
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:60]: This entire thing can be done in single statement.

File: services/audit-service/src/utils/file-check.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  if (customFilter.date == null || filterUsed.date == null) {
+    return true;
+  }
+  if (
+    filterUsed.date.fromDate <= customFilter.date.toDate &&
+    filterUsed.date.toDate >= customFilter.date.fromDate
+  ) {
+    return true;
+  }
+  return false;
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:76]: This can be optimized further

File: services/audit-service/src/utils/file-check.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import {CustomFilter} from '../models';
+
+export function checkActedOn(
+  filterUsed: CustomFilter,
+  customFilter: CustomFilter,
+): boolean {
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:6]: optimize these if-else blocks

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/audit-service/src/utils/file-check.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  const customActedOnList = customFilter.actedOnList;
+  // Check if both actedOn and actedOnList are null or if customFilter's actedOnList is null
+  // Check if filterUsed.actedOn is defined and included in customFilter.actedOnList
+  // Check if both actedOnLists have common elements
+  return (
+    (actedOn == null && actedOnList == null) ||
+    customActedOnList == null ||
+    (actedOn && customActedOnList.includes(actedOn)) ||
+    (actedOnList && haveCommonElements(customActedOnList, actedOnList))
+  );
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts]:
```suggestion
  const {actedOn, actedOnList} = filterUsed;
  const customActedOnList = customFilter.actedOnList;
  actedOnList = actedOnList ?? [];
  
  if (!!actedOn) {
    actedOnList = [...new Set([...actedOnList, actedOn])];
  }
  
  return (
    !customActedOnList ||
    actedOnList.length === 0 || 
    haveCommonElements(customActedOnList, actedOnList)
  );
```

File: services/audit-service/src/utils/file-check.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  filterUsed: CustomFilter,
+  customFilter: CustomFilter,
+): boolean {
+  const actionGroupList = filterUsed.actionGroupList;
+  const customActionGroupList = customFilter.actionGroupList;
+  return (
+    customActionGroupList == null ||
+    actionGroupList == null ||
+    haveCommonElements(actionGroupList, customActionGroupList)
+  );
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:31]:
```suggestion
  const actionGroupList = filterUsed.actionGroupList;
  const customActionGroupList = customFilter.actionGroupList;
  return (
    !customActionGroupList ||
    !actionGroupList ||
    haveCommonElements(actionGroupList, customActionGroupList)
  );
```

File: services/audit-service/src/utils/file-check.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export function checkEntityId(
+  filterUsed: CustomFilter,
+  customFilter: CustomFilter,
+): boolean {
+  return (
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:38]: Do the same changes here too

File: services/audit-service/src/utils/file-check.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  );
+}
+
+export function checkDates(
+  filterUsed: CustomFilter,
+  customFilter: CustomFilter,
+) {
+  const {fromDate: usedFromDate, toDate: usedToDate} = filterUsed.date!;
+  const {fromDate: customFromDate, toDate: customToDate} = customFilter.date!;
+  return (
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:54]: same changes

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/audit-service/src/utils/file-check.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  // Check if both actedOnLists have common elements
+  if (!!actedOn) {
+    actedOnList = [...new Set([...actedOnList, actedOn])];
+  }
+
+  return (
+    (actedOn == null && actedOnList == null) ||
+    customActedOnList == null ||
+    haveCommonElements(customActedOnList, actedOnList)
+  );
```
Comment (samarpan-b) [inline services/audit-service/src/utils/file-check.ts:21]:
```suggestion
  return (
    !customActedOnList || actedOnList.length === 0 ||
    haveCommonElements(customActedOnList, actedOnList)
  );
```

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/audit-service/src/utils/file-check.ts (hunk 6 above)
Comment (yeshamavani) [reply services/audit-service/src/utils/file-check.ts]: done

File: services/audit-service/src/utils/file-check.ts (hunk 7 above)
Comment (yeshamavani) [reply services/audit-service/src/utils/file-check.ts:41]: done

File: services/audit-service/src/utils/file-check.ts (hunk 8 above)
Comment (yeshamavani) [reply services/audit-service/src/utils/file-check.ts:52]: done

**Pull request merged**